.github/workflows/hrm-ecs-export.yml

```diff
@@ -0,0 +1,136 @@
+# Copyright (C) 2021-2023 Technology Matters
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see https://www.gnu.org/licenses/.
+
+name: HRM Export and (Re)populate the S3 bucket
+
+on:
+  workflow_dispatch:
+    secrets:
+      AWS_ACCESS_KEY_ID:
+        required: true
+      AWS_SECRET_ACCESS_KEY:
+        required: true
+    inputs:
+      start-date:
+        description: 'Start date for the S3 population (YYYY-MM-DD)'
+        required: true
+      end-date:
+        description: 'End date for the S3 population (YYYY-MM-DD)'
+        required: true
+      environment:
+        description: HRM environment to deploy.
+        default: development
+        required: true
+        type: choice
+        options:
+          - development
+          - staging
+          - production
+      region:
+        description: AWS region to update. E.G = us-east-1, eu-west-1 (must match with the AWS environment value). Default value = us-east-1
+        type: string
+        default: us-east-1
+        required: true
+      send-slack-message:
+        description: 'Specifies if should send a Slack message at the end of successful run. Defaults to true'
+        required: false
+        default: 'true'
+        type: string
+
+env:
+  ECR_REPOSITORY: ${{ inputs.environment }}-hrm
+
+jobs:
+  execute-ecs-task:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ inputs.region }}
+          # this plugin sets the AWS account ID to a secret which is not allowed in outputs
+          # we have to disable that so env.DOCKER_IMAGE will work
+          mask-aws-account-id: 'no'
+      
+      # Run the task definition and add to env
+      - name: Run Task Definition
+        run: |
+          task_definition=$(aws ecs describe-task-definition --task-definition ${{ env.ECR_REPOSITORY }}-task --query taskDefinition.taskDefinitionArn)
```

Comment:
`taskDefinition.taskDefinitionArn` is querying only the taskDefinitionArn in the above query.

You can test test these queries locally by installing aws-cli and jq or by getting a shell on the terraform container by running `make shell` from any valid terraform/terragrunt directory.

I would suggest that you debug these jq queries/bash chunks locally before trying to run the script.

How much bash/JQ do you want to learn? I can see several problems, but am not sure how much you'll learn if I just tell you the solutions as opposed to letting you debug via trial and error so you actually understand what jq is doing.